### PR TITLE
[extractor/epoch] Add extractor

### PR DESCRIPTION
**Pull request by tejasa97**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

Added a new extractor for EpochTV. Issue: https://github.com/yt-dlp/yt-dlp/issues/4714

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by tejasa97** [comment]
For issue: https://github.com/yt-dlp/yt-dlp/issues/4714

File: yt_dlp/YoutubeDL.py [hunk 1]
```diff
@@ -2404,6 +2404,7 @@ def process_video_result(self, info_dict, download=True):
         assert info_dict.get('_type', 'video') == 'video'
         self._num_videos += 1
 
+        print('info dict', info_dict)
```
Comment (HobbyistDev) [inline yt_dlp/YoutubeDL.py:2407]: ?

File: yt_dlp/extractor/_extractors.py [hunk 2]
```diff
@@ -1797,6 +1797,7 @@
 from .trueid import TrueIDIE
 from .trunews import TruNewsIE
 from .truth import TruthIE
+from .epoch import EpochIE
```
Comment (HobbyistDev) [inline yt_dlp/extractor/_extractors.py:1800]: Sort

File: yt_dlp/extractor/epoch.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+from .common import InfoExtractor
+from ..utils import (
+    unescapeHTML
+)
```
Comment (HobbyistDev) [inline yt_dlp/extractor/epoch.py:4]:
```suggestion
from ..utils import unescapeHTML
```

File: yt_dlp/extractor/epoch.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                'extractor_key': 'Epoch'
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._search_regex(r'<title>(?P<title>.*)<\/title>', unescapeHTML(webpage), 'title', group='title')
```
Comment (HobbyistDev) [inline yt_dlp/extractor/epoch.py:52]: Why didn't just use `html_extract_title` ?

File: yt_dlp/extractor/epoch.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        youmaker_video_id = self._search_regex(
+            r'<div class="videobox" id="videobox" data-trailer="[\w-]+" data-id="([\w-]+)">', webpage, 'url')
+
+        return {
+            "id": video_id,
+            "ext": "mp4",
+            "url": f"http://vs1.youmaker.com/assets/{youmaker_video_id}/playlist.m3u8",
```
Comment (HobbyistDev) [inline yt_dlp/extractor/epoch.py:59]: Use single quote
```suggestion
            'id': video_id,
            'ext': 'mp4',
            'url': f'http://vs1.youmaker.com/assets/{youmaker_video_id}/playlist.m3u8',
```

File: yt_dlp/extractor/epoch.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+
+class EpochIE(InfoExtractor):
+    _VALID_URL = r'https?://www.theepochtimes\.com/[\w-]+_(?P<id>\d+).html'
+    _TESTS = [
+        {
+            'url': 'https://www.theepochtimes.com/they-can-do-audio-video-physical-surveillance-on-you-24h-365d-a-year-rex-lee-on-intrusive-apps_4661688.html',
+            'info_dict': {
+                "id": '4661688',
```
Comment (HobbyistDev) [inline yt_dlp/extractor/epoch.py:13]:
```suggestion
                'id': '4661688',
```

File: yt_dlp/extractor/epoch.py (hunk 4 above)
Comment (HobbyistDev) [reply yt_dlp/extractor/epoch.py:52]: Inline `title` in return

File: yt_dlp/extractor/epoch.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+                'ext': 'mp4',
+                'url': 'http://vs1.youmaker.com/assets/a3dd732c-4750-4bc8-8156-69180668bda1/playlist.m3u8',
+                'title': '‘They Can Do Audio, Video, Physical Surveillance on You 24H/365D a Year’: Rex Lee on Intrusive Apps',
+                'webpage_url_domain': 'theepochtimes.com',
+                'extractor': 'Epoch',
+                'extractor_key': 'Epoch'
```
Comment (pukkandan) [inline yt_dlp/extractor/epoch.py:19]:
```suggestion
                'title': '‘They Can Do Audio, Video, Physical Surveillance on You 24H/365D a Year’: Rex Lee on Intrusive Apps',
```
Same below

File: yt_dlp/extractor/epoch.py (hunk 5 above)
Comment (pukkandan) [inline yt_dlp/extractor/epoch.py:59]: Use `_extract_m3u8_formats_and_subtitles`

File: yt_dlp/extractor/epoch.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        },
+    ]
+
+    def _real_extract(self, url):
+
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = self._search_regex(r'<title>(?P<title>.*)<\/title>', unescapeHTML(webpage), 'title', group='title')
+        youmaker_video_id = self._search_regex(
+            r'<div class="videobox" id="videobox" data-trailer="[\w-]+" data-id="([\w-]+)">', webpage, 'url')
```
Comment (pukkandan) [inline yt_dlp/extractor/epoch.py:54]: Make regex more relaxed

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/epoch.py (hunk 7 above)
Comment (tejasa97) [reply yt_dlp/extractor/epoch.py:19]: sorry i'm a little confused. do you mean fields `url`, `webpage_url_domain`, `extractor`, `extractor_key` aren't necessary?

Comment (pukkandan) [reply yt_dlp/extractor/epoch.py:19]: Yes, they shouldn't be used in tests

File: yt_dlp/extractor/epoch.py (hunk 8 above)
Comment (tejasa97) [reply yt_dlp/extractor/epoch.py]: @pukkandan please check now

**Pull request merged**